data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaCustomConsumer.java

```diff
@@ -424,16 +426,25 @@ private void processRecord(final AcknowledgementSet acknowledgementSet, final Re
                 bufferAccumulator.add(record);
                 break;
             } catch (Exception e) {
+                if (!paused) {
+                    paused = true;
+                    consumer.pause(consumer.assignment());
+                }
                 if (e instanceof SizeOverflowException) {
                     topicMetrics.getNumberOfBufferSizeOverflows().increment();
                 } else {
                     LOG.debug("Error while adding record to buffer, retrying ", e);
                 }
                 try {
                     Thread.sleep(100);
+                    consumeRecords();
                 } catch (Exception ex) {} // ignore the exception because it only means the thread slept for shorter time
             }
         }
+        if (paused) {
+            consumer.resume(consumer.assignment());
```

Comment:
I think there may be an issue here. We may have to call resume on exactly the same partitions that we paused. e.g. lets say consumer initially had 4 partitions that were paused. While paused, 2 got revoked and we resumed only 2 assigned partitions. Now same 2 partitions are assigned back, will they remain paused? because we never called resume on them. It'll be good to test.

Reply:
If the partitions have moved out of the consumer, I do not think they will be in the paused state. I do not think we can resume the partitions that are currently not owned by a consumer. But will try to test it.

Reply:
I have tested this scenario. When the partitions are assigned back, they are not paused. I think once the partitions are revoked, the partitions are removed from the consumer's assignment. When they are assigned back, they are assigned as any other new partition. There is no stale state from previous assignment.